src/main.rs

```diff
@@ -909,7 +826,18 @@ fn probes_filter(probes: &[DebugProbeInfo], selector: &ProbeFilter) -> Vec<Debug
         .collect()
 }
 
-fn print_probes(probes: Vec<DebugProbeInfo>) -> Result<i32, anyhow::Error> {
+fn print_chips() -> i32 {
```

Comment:
Let's either return an `anyhow::Result<()>` or nothing from these helpers. We should only deal with raw exit codes when necessary, since they're quite non-Rusty.

Reply:
And then `match` it in `fn main`?

This makes it more complex, but we could also let `fn notmain` return `anyhow::Result<Option<i32>>` and then just pass on `None` and `process::exit` on `Some(...)`?

Reply:
If these all return `()`, `notmain` can just always return `Ok(0)`